logrus.go

```diff
@@ -113,6 +114,7 @@ type FieldLogger interface {
 	WithField(key string, value interface{}) *Entry
 	WithFields(fields Fields) *Entry
 	WithError(err error) *Entry
+	WithTime(t time.Time) *Entry
```

Comment:
@sbrisson2 is this interface change really needed for your use case. I'm kind of worried it could cause in any of the thousands importing package. It we can remove this interface change, it is fine with me. If you really need this method in an interface, you could declare in another interface extending this one. Sorry for the delay.

Reply:
I don't need the interface change.  I can just add the methods in the `Logger` and `Entry` types.  I'll update the PR.